aiosmtpd/smtp.py

```diff
@@ -78,8 +78,9 @@ def __init__(self, handler,
                  loop=None):
         self.__ident__ = ident or __ident__
         self.loop = loop if loop else make_loop()
+        limit = 1001  # TODO compute properly and make configurable
```

Comment:
I'd really love to see this TODO implemented. I'm not sure what the best approach is here. On [this PR](https://github.com/aio-libs/aiosmtpd/pull/146#issuecomment-396927244) @asvetlov recommended the use of http://www.attrs.org/en/stable/

Reply:
This number is a bit hard to tune. If too small, we run the risk of StreamReader raising LimitOverrunError regularly if the client is a bit "eager". Too big and we're wasting space.

Need to profile the throughput of _handle_client w.r.t. StreamReader.

I'd say 1001 is waaay to small, though :-)

Reply:
Okay, so I misunderstood what "limit" meant in StreamReader. I'm good with 1001.

Reply:
I've resolved this. The line length is now configurable, and also references to RFC 5321 have been added to the source code, the error message, and `smtp.rst`